### PR TITLE
refactor(Keychain)!: proper naming for BIP44 and DIP9 get keys methods

### PR DESCRIPTION
**Pull request by Alex-Werner**
## Issue being fixed or feature implemented

This breaking change PR bring a small naming update, indeed we had getHardenedBIP44Path and getDIP9FeaturePath methods that directly returns keys. 
As naming is documentation by itself, this modification will better reflect expected comportment of those methods. 

## What was done?
- renamed getHardenedBIP44Path to getHardenedBIP44HDKey
- renamed getHardenedDIP9FeaturePath to getHardenedDIP9FeatureHDKey
- added missing test for getHardenedDIP9FeatureHDKey

## How Has This Been Tested?
- Added new test

## Breaking Changes
-  `getHardenedBIP44Path` renamed to `getHardenedBIP44HDKey`
- `getHardenedDIP9FeaturePath` renamed to  `getHardenedDIP9FeatureHDKey`

## Checklist:

- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have added or updated relevant unit/integration/functional/e2e tests
- [X] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [X] I have assigned this pull request to a milestone

### CODE REVIEW

**Review by antouhou — APPROVED** [review]
Please be very careful with merging this one. First, create an SDK branch with wallet-lib installed from this branch. Then use this new branch of SDK in the dashmate and test-suite. When the test suite PR is green, you can start merging it into the v21

**Comment by Alex-Werner** [comment]
Yes, it would not be something we can just merge like that. 
I did perform analysis of usage, we actually rarely use it, so there is that point, however, it remain a breaking change for a low value item. 
I was thinking of waiting for other breaking changes to be merged (such as #291) before merging this one. 

Will follow the plan !

**Review by shumkov — APPROVED** [review]
LGTM! 👍

**Pull request merged**
